.github/workflows/pulls.yml

```diff
@@ -0,0 +1,85 @@
+name: Test All Grammar
+
+on:
+  pull_request:
+    branches:
+      - "main"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    outputs:
+      self_mutation_happened: ${{ steps.self_mutation.outputs.self_mutation_happened }}
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+        with:
+          node-version-file: '.nvmrc'
+      - name: Install dependencies
+        run: npm ci
+      - name: build
+        run: npm run build
```

Comment:
Since we're now running Go in CI, should we add a `setup-go` action and maybe also a `.go-version` file, to make this more reproducible?

```yaml
      - name: Set up Go
        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
        with:
          go-version-file: ".go-version"
```

Reply:
Sounds good! I set go.mod as the version file as it already exists and has a version constraint in there.

Reply:
Did you push that change?